src/utilities.jl

```diff
@@ -30,47 +30,92 @@ function get_word(tdpp::TextDocumentPositionParams, server::LanguageServerInstan
     return String(word)
 end
 
-function get_sym(str::AbstractString)
-    name = split(str, '.')
-    try
-        x = getfield(Main, Symbol(name[1]))
-        for i = 2:length(name)
-            x = getfield(x, Symbol(name[i]))
+
+function unpack_dot(id, args = Symbol[])
+    if id isa Expr && id.head == :. && id.args[2] isa QuoteNode
+        if id.args[2].value isa Symbol
+            unshift!(args, id.args[2].value)
+            unpack_dot(id.args[1], args)
+        else
+            return Symbol[]
         end
-        return x
-    catch
-        return nothing
+    elseif id isa Symbol
+        unshift!(args, id)
+    else
+        return Symbol[]
     end
+    return args
 end
 
-function get_cache_entry(word, server, modules=[])
-    allmod = vcat([:Base, :Core], modules)
-    entry = (:EMPTY, "", [])
-    if search(word, ".")!=0:-1
-        sword = split(word, ".")
-        modname = parse(join(sword[1:end-1], "."))
-        if Symbol(first(sword)) in allmod && modname in keys(server.cache) && Symbol(last(sword)) in keys(server.cache[modname])
-            entry = server.cache[modname][Symbol(last(sword))]
-        end
+
+repack_dot(args::Symbol) = args
+function repack_dot(args::Vector)
+    if length(args) == 1
+        return first(args)
+    else
+        return repack_dot([Expr(:., first(args), QuoteNode(args[2])); args[3:end]])
+    end
+end
+
+
+function get_module(ids::Vector{Symbol}, M = Main)
+    if isempty(ids)
+        return M
+    elseif isdefined(M, first(ids))
+        M = getfield(M, shift!(ids))
+        return get_module(ids, M)
     else
-        for m in allmod
-            if m in keys(server.cache) && Symbol(word) in server.cache[m][:EXPORTEDNAMES]
-                entry = server.cache[m][Symbol(word)]
+        return false
+    end
+end
+
+
+function get_cache_entry(id, server, modules = [])
```

Comment:
How about make that change for a follow up PR? Would be great to first get this merged in a workable state. Perf optimizations can come later?